### ISSUE TITLE
COMException in FormDesignerRefactorRenameCommand

### ISSUE THREAD

**Issue by comintern** [rubberduck-vba/Rubberduck#4273]
Found this in the log file posted on #4202:

```
2018-07-17 16:24:49.7148;ERROR-2.2.0.3508;Rubberduck.UI.Command.Refactorings.FormDesignerRefactorRenameCommand;System.Runtime.InteropServices.COMException (0x800A09A3): Exception from HRESULT: 0x800A09A3
   at Microsoft.Vbe.Interop._VBComponent.get_Designer()
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBComponent.get_HasDesigner() in C:\projects\rubberduck\Rubberduck.VBEditor.VBA\SafeComWrappers\VB\VBComponent.cs:line 85
   at Rubberduck.UI.Command.Refactorings.FormDesignerRefactorRenameCommand.GetTarget(Nullable`1 qualifiedModuleName) in C:\projects\rubberduck\Rubberduck.Core\UI\Command\Refactorings\FormDesignerRefactorRenameCommand.cs:line 62
   at Rubberduck.UI.Command.Refactorings.FormDesignerRefactorRenameCommand.EvaluateCanExecute(Object parameter) in C:\projects\rubberduck\Rubberduck.Core\UI\Command\Refactorings\FormDesignerRefactorRenameCommand.cs:line 29
   at Rubberduck.UI.Command.CommandBase.CanExecute(Object parameter) in C:\projects\rubberduck\Rubberduck.Core\UI\Command\CommandBase.cs:line 28;System.Runtime.InteropServices.COMException (0x800A09A3): Exception from HRESULT: 0x800A09A3
   at Microsoft.Vbe.Interop._VBComponent.get_Designer()
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBComponent.get_HasDesigner() in C:\projects\rubberduck\Rubberduck.VBEditor.VBA\SafeComWrappers\VB\VBComponent.cs:line 85
   at Rubberduck.UI.Command.Refactorings.FormDesignerRefactorRenameCommand.GetTarget(Nullable`1 qualifiedModuleName) in C:\projects\rubberduck\Rubberduck.Core\UI\Command\Refactorings\FormDesignerRefactorRenameCommand.cs:line 62
   at Rubberduck.UI.Command.Refactorings.FormDesignerRefactorRenameCommand.EvaluateCanExecute(Object parameter) in C:\projects\rubberduck\Rubberduck.Core\UI\Command\Refactorings\FormDesignerRefactorRenameCommand.cs:line 29
   at Rubberduck.UI.Command.CommandBase.CanExecute(Object parameter) in C:\projects\rubberduck\Rubberduck.Core\UI\Command\CommandBase.cs:line 28
```
This error shows up repeatedly in the log file, and appears to be the cause of 
daFreeMan's second comment on that issue.

It may also be related to this error from the same log, but that may require further investigation:

```
2018-07-17 16:24:48.8538;WARN-2.2.0.3508;Rubberduck.Inspections.Rubberduck.Inspections.Inspector;System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at Rubberduck.Inspections.Abstract.InspectionBase.GetInspectionResults(CancellationToken token) in C:\projects\rubberduck\Rubberduck.CodeAnalysis\Inspections\Abstract\InspectionBase.cs:line 168
   at Rubberduck.Inspections.Rubberduck.Inspections.Inspector.RunInspection(IInspection inspection, ConcurrentBag`1 allIssues, CancellationToken token) in C:\projects\rubberduck\Rubberduck.CodeAnalysis\Inspections\Inspector.cs:line 187;System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at Rubberduck.Inspections.Abstract.InspectionBase.GetInspectionResults(CancellationToken token) in C:\projects\rubberduck\Rubberduck.CodeAnalysis\Inspections\Abstract\InspectionBase.cs:line 168
   at Rubberduck.Inspections.Rubberduck.Inspections.Inspector.RunInspection(IInspection inspection, ConcurrentBag`1 allIssues, CancellationToken token) in C:\projects\rubberduck\Rubberduck.CodeAnalysis\Inspections\Inspector.cs:line 187
```
A Google search for "HRESULT 0x800A09A3" only returned the RD issue #4202 (now there will be 2 results...), but searching for "0x800A09A3" turned up [this interesting tidbit](https://www.devexpress.com/Support/Center/Question/Details/T329338/import-reports-from-ms-access)...

> The expression you entered refers to an object that is closed or doesn't exist.

...that makes me think RD is messing with something that is data bound in a way that it shouldn't be.

**Comment by bclothier** [comment]
FYI the `The expression you entered refers to an object that is closed or doesn't exist` is a fairly common error in Access, typically associated with trying to access an object that was loaded previously but has became unloaded (e.g. trying to reach inside a subform while in middle of closing of the parent form). 

I have a hunch that the quickfix is trying to change the code module in an open Access form. Some changes are legal but some will require a reset of the code. Besides getting the designer wouldn't be appropriate in that case since there's no designer (at least not within the VBIDE OM) for a document module.

**Comment by bclothier** [comment]
Tested with the latest build and cannot replicate this. Seems to have been fixed by #3389.
